handler/PostMessageSentHandler.ts

```diff
@@ -92,6 +92,7 @@ export class PostMessageSentHandler {
             response = (await Dialogflow.sendRequest(this.http, this.read, this.modify, rid, text, DialogflowRequestType.MESSAGE));
         } catch (error) {
             this.app.getLogger().error(`${Logs.DIALOGFLOW_REST_API_ERROR} ${error.message}`);
+            console.error(`${Logs.DIALOGFLOW_REST_API_ERROR} ${error.message}`);
```

Comment:
@ItzzRitik is it possible to json.stringify this error message?  We want to work around the multi-line logging issue when viewing logs in kibana.  Thanks.

Reply:
@Shailesh351 and Eric, Please tell me if I'm wrong,
I think **error.message** is in string format by default.
have we overridden our **error.message** data type by any means?

Reply:
That depends on what we are getting in the `error` object we are catching. We can use `getError` fn to always log string. If input is string it returns a string, if input is the object it returns JSON.stringify(object).

Reply:
@ItzzRitik should we also add console.error to the error on line 165?  In general, can you find all spots where DIALOGFLOW_REST_API_ERROR is used and make sure we include that in the console.error?  Thanks.

Reply:
@Shailesh351 @ItzzRitik FWIW - this is what I am seeing in kibana for this log point:
"Error occurred while using Dialogflow Rest API. Details:- Error occurred while sending a HTTP Request: Error: An Error occurred while connecting to Dialogflow's REST API            Error Details:-"

And then later I can see the first line of the 'Error Details:-', as if it were a new multiline message.

Reply:
Sure Eric, I'll add console.error for all the occurrences of DIALOGFLOW_REST_API_ERROR.
Also Can you please attach a screenshot of the error having 'Error Details:-'
it will be easier to understand which part of the output, we should add JSON.Stringify.

Reply:
@ItzzRitik the 'Error Details:-' section comes in as a lot of multiline log points......  but the entire details looks like this:  

Error Details:-\n                message:- com.google.apps.framework.request.BadRequestException: More than 10 nested flow transitions detected: \n[{\n  \"Step 1\": {\n    \"Type\": \"INITIAL_STATE\",\n    \"InitialState\": {\n      \"MatchedIntent\": {\n        \"Id\": \"408d2eff-7e04-40a8-a427-ad51bc187d49\",\n        \"DisplayName\": \"Usage\",\n        \"Type\": \"NLU\",\n        \"Score\": 1.0,\n        \"Active\": true\n      },\n      \"FlowState\": {\n        \"Name\": \"Default Start Flow\",\n        \"FlowId\": \"00000000-0000-0000-0000-000000000000\",\n        \"PageState\": {\n          \"Name\": \"Main Menu\",\n          \"PageId\": \"9f72b5c0-0383-434f-9cf9-70e097111e58\",\n          \"Status\": \"TRANSITION_ROUTING\",\n          \"FormFilled\": true\n        },\n        \"Version\": 0.0\n      },\n      \"SessionParameters\": {\n        \"billingOptions\": \"Boleto\",\n        \"username\": \"16177808477\",\n        \"wh_account_buy_more_available\": true,\n        \"wh_account_is_unlimited\": false,\n        \"wh_account_status\": \"ACTIVE\",\n        \"wh_account_valid\": 1.0,\n        \"wh_account_vde_status\": true,\n        \"wh_invoices_amount\": \"\",\n        \"wh_invoices_count\": 0.0,\n        \"wh_invoices_date\": \"\",\n        \"wh_invoices_id\": \"\",\n        \"wh_plan_allowed_gb\": 25.0,\n        \"wh_plan_bought_more_gb\": 0.0,\n        \"wh_plan_current_gb\": 0.0,\n        \"wh_plan_data_allowance_policy\": \"After Priority Data, 1-5 Mbps download speeds. Faster speeds in the morning/daytime. Slower speeds in evening.\",\n        \"wh_plan_name\": \"Liberty 25 - WiFi Incl - LF - FZ\",\n        \"wh_plan_reset_date\": {\n          \"day\": 16.0,\n          \"hours\": null,\n          \"minutes\": null,\n          \"month\": 9.0,\n          \"nanos\": null,\n          \"seconds\": null,\n          \"year\": 2021.0\n        },\n        \"wh_plan_video_quality\": \"The Video Data Extender saves your data by streaming video at DVD quality, optimized for 480p.\"\n      }\n    }\n  }\n}, {\n  \"Step 2\": {\n    \"Type\": \"STATE_MACHINE\",\n    \"StateMachine\": {\n      \"FlowState\": {\n        \"Name\": \"Default Start Flow\",\n        \"FlowId\": \"00000000-0000-0000-0000-000000000000\",\n        \"PageState\": {\n          \"Name\": \"Main Menu\",\n          \"PageId\": \"9f72b5c0-0383-434f-9cf9-70e097111e58\",\n          \"Status\": \"TRANSITION_ROUTING\",\n          \"FormFilled\": true\n        },\n        \"Version\": 0.0\n      },\n      \"TriggeredIntent\": \"Usage\",\n      \"TriggeredTransitionRouteId\": \"404bc1d4-d83a-4615-8b87-2b8eb096678c\",\n      \"TargetFlow\": \"457e5ac5-60c1-4cce-8327-e22564fed084\"\n    },\n    \"FunctionExecution\": {\n      \"Responses\": [],\n      \"Webhook\": {\n        \"Status\": \"OK\",\n        \"Latency\": \"176 ms\"\n      }\n    }\n  }\n}, {\n  \"Step 3\": {\n    \"Type\": \"STATE_MACHINE\",\n    \"StateMachine\": {\n      \"FlowState\": {\n        \"Name\": \"Usage\",\n        \"FlowId\": \"457e5ac5-60c1-4cce-8327-e22564fed084\",\n        \"PageState\": {\n          \"Name\": \"Start Page\",\n          \"PageId\": \"START_PAGE\",\n          \"Status\": \"TRANSITION_ROUTING\"\n        },\n        \"Version\": 0.0\n      },\n      \"TriggeredIntent\": \"Usage\",\n      \"TriggeredTransitionRouteId\": \"756a931d-dcd8-498c-9c1e-82b056945828\",\n      \"TargetPage\": \"55ee566a-f989-4b79-94ea-4ea95cfaacfe\"\n    },\n    \"FunctionExecution\": {\n      \"Responses\": []\n    }\n  }\n}, {\n  \"Step 4\": {\n    \"Type\": \"STATE_MACHINE\",\n    \"StateMachine\": {\n      \"FlowState\": {\n        \"Name\": \"Usage\",\n        \"FlowId\": \"457e5ac5-60c1-4cce-8327-e22564fed084\",\n        \"PageState\": {\n          \"Name\": \"Get Usage\",\n          \"PageId\": \"55ee566a-f989-4b79-94ea-4ea95cfaacfe\",\n          \"Status\": \"ENTERING_PAGE\"\n        },\n        \"Version\": 0.0\n      }\n    },\n    \"FunctionExecution\": {\n      \"Responses\": [],\n      \"Webhook\": {\n        \"Status\": \"OK\",\n        \"Latency\": \"108 ms\"\n      }\n    }\n  }\n}, {\n  \"Step 5\": {\n    \"Type\": \"STATE_MACHINE\",\n    \"StateMachine\": {\n      \"FlowState\": {\n        \"Name\": \"Usage\",\n        \"FlowId\": \"457e5ac5-60c1-4cce-8327-e22564fed084\",\n        \"PageState\": {\n          \"Name\": \"Get Usage\",\n          \"PageId\": \"55ee566a-f989-4b79-94ea-4ea95cfaacfe\",\n          \"Status\": \"TRANSITION_ROUTING\",\n          \"FormFilled\": true\n        },\n        \"Version\": 0.0\n      },\n      \"TriggeredTransitionRouteId\": \"c49df739-67d9-47ce-930d-9304233fe9c8\",\n      \"TriggeredCondition\": \"true\",\n      \"TargetPage\": \"255db95b-9898-4ad5-9e1d-c0ebed877c99\"\n    },\n    \"FunctionExecution\": {\n      \"Responses\": []\n    }\n  }\n}, {\n  \"Step 6\": {\n    \"Type\": \"STATE_MACHINE\",\n    \"StateMachine\": {\n      \"FlowState\": {\n        \"Name\": \"Usage\",\n        \"FlowId\": \"457e5ac5-60c1-4cce-8327-e22564fed084\",\n        \"PageState\": {\n          \"Name\": \"Display Usage\",\n          \"PageId\": \"255db95b-9898-4ad5-9e1d-c0ebed877c99\",\n          \"Status\": \"ENTERING_PAGE\"\n        },\n        \"Version\": 0.0\n      }\n    },\n    \"FunctionExecution\": {\n      \"Responses\": [{\n        \"text\": {\n          \"text\": [\"Você já utlizou 0 GB do seu pacote de dados mensal. Sua franquia mensal do seu plano Liberty 25 - WiFi Incl - LF - FZ é 25 GB e será reiniciado em 16/09/2021.\"],\n          \"redactedText\": [\"Você já utlizou 0 GB do seu pacote de dados mensal. Sua franquia mensal do seu plano Liberty 25 - WiFi Incl - LF - FZ é 25 GB e será reiniciado em 16/09/2021.\"]\n        },\n        \"responseType\": \"ENTRY_PROMPT\",\n        \"source\": \"VIRTUAL_AGENT\"\n      }],\n      \"SystemFunctionResults\": {\n        \"$sys.func.FORMAT_DATE\": [\"16/09/2021\"]\n      }\n    }\n  }\n}]                status:- com.google.apps.framework.request.BadRequestException: More than 10 nested flow transitions detected: \n[{\n  \"Step 1\": {\n    \"Type\": \"INITIAL_STATE\",\n    \"InitialState\": {\n      \"MatchedIntent\": {\n        \"Id\": \"408d2eff-7e04-40a8-a427-ad51bc187d49\",\n        \"DisplayName\": \"Usage\",\n        \"Type\": \"NLU\",\n        \"Score\": 1.0,\n        \"Active\": true\n      },\n      \"FlowState\": {\n        \"Name\": \"Default Start Flow\",\n        \"FlowId\": \"00000000-0000-0000-0000-000000000000\",\n        \"PageState\": {\n          \"Name\": \"Main Menu\",\n          \"PageId\": \"9f72b5c0-0383-434f-9cf9-70e097111e58\",\n          \"Status\": \"TRANSITION_ROUTING\",\n          \"FormFilled\": true\n        },\n        \"Version\": 0.0\n      },\n      \"SessionParameters\": {\n        \"billingOptions\": \"Boleto\",\n        \"username\": \"16177808477\",\n        \"wh_account_buy_more_available\": true,\n        \"wh_account_is_unlimited\": false,\n        \"wh_account_status\": \"ACTIVE\",\n        \"wh_account_valid\": 1.0,\n        \"wh_account_vde_status\": true,\n        \"wh_invoices_amount\": \"\",\n        \"wh_invoices_count\": 0.0,\n        \"wh_invoices_date\": \"\",\n        \"wh_invoices_id\": \"\",\n        \"wh_plan_allowed_gb\": 25.0,\n        \"wh_plan_bought_more_gb\": 0.0,\n        \"wh_plan_current_gb\": 0.0,\n        \"wh_plan_data_allowance_policy\": \"After Priority Data, 1-5 Mbps download speeds. Faster speeds in the morning/daytime. Slower speeds in evening.\",\n        \"wh_plan_name\": \"Liberty 25 - WiFi Incl - LF - FZ\",\n        \"wh_plan_reset_date\": {\n          \"day\": 16.0,\n          \"hours\": null,\n          \"minutes\": null,\n          \"month\": 9.0,\n          \"nanos\": null,\n          \"seconds\": null,\n          \"year\": 2021.0\n        },\n        \"wh_plan_video_quality\": \"The Video Data Extender saves your data by streaming video at DVD quality, optimized for 480p.\"\n      }\n    }\n  }\n}, {\n  \"Step 2\": {\n    \"Type\": \"STATE_MACHINE\",\n    \"StateMachine\": {\n      \"FlowState\": {\n        \"Name\": \"Default Start Flow\",\n        \"FlowId\": \"00000000-0000-0000-0000-000000000000\",\n        \"PageState\": {\n          \"Name\": \"Main Menu\",\n          \"PageId\": \"9f72b5c0-0383-434f-9cf9-70e097111e58\",\n          \"Status\": \"TRANSITION_ROUTING\",\n          \"FormFilled\": true\n        },\n        \"Version\": 0.0\n      },\n      \"TriggeredIntent\": \"Usage\",\n      \"TriggeredTransitionRouteId\": \"404bc1d4-d83a-4615-8b87-2b8eb096678c\",\n      \"TargetFlow\": \"457e5ac5-60c1-4cce-8327-e22564fed084\"\n    },\n    \"FunctionExecution\": {\n      \"Responses\": [],\n      \"Webhook\": {\n        \"Status\": \"OK\",\n        \"Latency\": \"176 ms\"\n      }\n    }\n  }\n}, {\n  \"Step 3\": {\n    \"Type\": \"STATE_MACHINE\",\n    \"StateMachine\": {\n      \"FlowState\": {\n        \"Name\": \"Usage\",\n        \"FlowId\": \"457e5ac5-60c1-4cce-8327-e22564fed084\",\n        \"PageState\": {\n          \"Name\": \"Start Page\",\n          \"PageId\": \"START_PAGE\",\n          \"Status\": \"TRANSITION_ROUTING\"\n        },\n        \"Version\": 0.0\n      },\n      \"TriggeredIntent\": \"Usage\",\n      \"TriggeredTransitionRouteId\": \"756a931d-dcd8-498c-9c1e-82b056945828\",\n      \"TargetPage\": \"55ee566a-f989-4b79-94ea-4ea95cfaacfe\"\n    },\n    \"FunctionExecution\": {\n      \"Responses\": []\n    }\n  }\n}, {\n  \"Step 4\": {\n    \"Type\": \"STATE_MACHINE\",\n    \"StateMachine\": {\n      \"FlowState\": {\n        \"Name\": \"Usage\",\n        \"FlowId\": \"457e5ac5-60c1-4cce-8327-e22564fed084\",\n        \"PageState\": {\n          \"Name\": \"Get Usage\",\n          \"PageId\": \"55ee566a-f989-4b79-94ea-4ea95cfaacfe\",\n          \"Status\": \"ENTERING_PAGE\"\n        },\n        \"Version\": 0.0\n      }\n    },\n    \"FunctionExecution\": {\n      \"Responses\": [],\n      \"Webhook\": {\n        \"Status\": \"OK\",\n        \"Latency\": \"108 ms\"\n      }\n    }\n  }\n}, {\n  \"Step 5\": {\n    \"Type\": \"STATE_MACHINE\",\n    \"StateMachine\": {\n      \"FlowState\": {\n        \"Name\": \"Usage\",\n        \"FlowId\": \"457e5ac5-60c1-4cce-8327-e22564fed084\",\n        \"PageState\": {\n          \"Name\": \"Get Usage\",\n          \"PageId\": \"55ee566a-f989-4b79-94ea-4ea95cfaacfe\",\n          \"Status\": \"TRANSITION_ROUTING\",\n          \"FormFilled\": true\n        },\n        \"Version\": 0.0\n      },\n      \"TriggeredTransitionRouteId\": \"c49df739-67d9-47ce-930d-9304233fe9c8\",\n      \"TriggeredCondition\": \"true\",\n      \"TargetPage\": \"255db95b-9898-4ad5-9e1d-c0ebed877c99\"\n    },\n    \"FunctionExecution\": {\n      \"Responses\": []\n    }\n  }\n}, {\n  \"Step 6\": {\n    \"Type\": \"STATE_MACHINE\",\n    \"StateMachine\": {\n      \"FlowState\": {\n        \"Name\": \"Usage\",\n        \"FlowId\": \"457e5ac5-60c1-4cce-8327-e22564fed084\",\n        \"PageState\": {\n          \"Name\": \"Display Usage\",\n          \"PageId\": \"255db95b-9898-4ad5-9e1d-c0ebed877c99\",\n          \"Status\": \"ENTERING_PAGE\"\n        },\n        \"Version\": 0.0\n      }\n    },\n    \"FunctionExecution\": {\n      \"Responses\": [{\n        \"text\": {\n          \"text\": [\"Você já utlizou 0 GB do seu pacote de dados mensal. Sua franquia mensal do seu plano Liberty 25 - WiFi Incl - LF - FZ é 25 GB e será reiniciado em 16/09/2021.\"],\n          \"redactedText\": [\"Você já utlizou 0 GB do seu pacote de dados mensal. Sua franquia mensal do seu plano Liberty 25 - WiFi Incl - LF - FZ é 25 GB e será reiniciado em 16/09/2021.\"]\n        },\n        \"responseType\": \"ENTRY_PROMPT\",\n        \"source\": \"VIRTUAL_AGENT\"\n      }],\n      \"SystemFunctionResults\": {\n        \"$sys.func.FORMAT_DATE\": [\"16/09/2021\"]\n      }\n